packages/cli/src/Server.ts

```diff
@@ -1246,35 +1253,35 @@ class App {
 		this.app.get(
 			`/${this.restEndpoint}/node-parameter-options`,
 			ResponseHelper.send(
-				async (req: express.Request, res: express.Response): Promise<INodePropertyOptions[]> => {
+				async (req: NodeParameterOptionsRequest): Promise<INodePropertyOptions[]> => {
 					const nodeTypeAndVersion = JSON.parse(
-						`${req.query.nodeTypeAndVersion}`,
+						req.query.nodeTypeAndVersion,
 					) as INodeTypeNameVersion;
-					const path = req.query.path as string;
-					let credentials: INodeCredentials | undefined;
+
+					const { path, methodName } = req.query;
+
 					const currentNodeParameters = JSON.parse(
-						`${req.query.currentNodeParameters}`,
+						req.query.currentNodeParameters,
 					) as INodeParameters;
-					if (req.query.credentials !== undefined) {
-						credentials = JSON.parse(req.query.credentials as string);
-					}
-					const methodName = req.query.methodName as string;
 
-					const nodeTypes = NodeTypes();
+					let credentials: INodeCredentials | undefined;
+
+					if (req.query.credentials) {
+						credentials = JSON.parse(req.query.credentials);
+						// TODO UM: restrict user access to credentials he cannot use.
+					}
 
-					// @ts-ignore
 					const loadDataInstance = new LoadNodeParameterOptions(
 						nodeTypeAndVersion,
-						nodeTypes,
+						NodeTypes(),
 						path,
 						currentNodeParameters,
 						credentials,
 					);
 
 					const additionalData = await WorkflowExecuteAdditionalData.getBase(currentNodeParameters);
-					// TODO UM: restrict user access to credentials he cannot use.
-					// @ts-ignore
-					additionalData.userId = (req.user as User).id;
+
+					additionalData.userId = req.user.id;
```

Comment:
why we need to add this here?

Reply:
Not my work, but likely because of this:
https://github.com/n8n-io/n8n/pull/2697